ss_admin_console/main/tests/classes/FieldReferenceTest.cls

```diff
@@ -0,0 +1,96 @@
+    @IsTest
+    public inherited sharing class FieldReferenceTest {
+
+    @IsTest
+    static void testSchemaGetField() {
+        FieldReference field = FieldReference.build(Account.Name);
+
+        Test.startTest();
+        SObjectField sobjField = field.getField();
+        Test.stopTest();
+
+        System.assertEquals(Account.Name, sobjField, 'Method must return SObjectField');
+    }
+
+    @IsTest
+    static void testSchemaGetFrom() {
+        FieldReference field = FieldReference.build(Account.Name);
+        Account acc = new Account(Name = 'TestAcct');
+
+        Test.startTest();
+        Object obj = field.getFrom(acc);
+        Test.stopTest();
+
+        System.assertEquals(acc.Name, obj, 'Method must return field value of sobject');
+    }
+
+    @IsTest
+    static void testSchemaToString() {
+        FieldReference field = FieldReference.build(Account.Name);
+
+        Test.startTest();
+        String strField = field.toString();
+        Test.stopTest();
+
+        System.assertEquals(Account.Name.getDescribe().getName(), strField);
+    }
+
+    @IsTest
+    static void testSchemaInvalidField() {
+        Test.startTest();
+        try {
+            FieldReference field = FieldReference.build(Schema.Account.getSObjectType(), 'Invalid Field');
+            System.assert(false, 'IllegalArgumentException is expected here');
+        } catch (IllegalArgumentException e) {
+            System.assertEquals('Field is invalid', e.getMessage());
+        }
+        Test.stopTest();
+    }
+
+    @IsTest 
+    static void testChainedGetField() {
+        FieldReference field = FieldReference.build(Schema.Opportunity.getSObjectType(), 'Account.Owner.Name');
+        
+        Test.startTest();
+        SObjectField sobjField = field.getField();
+        Test.stopTest();
+
+        System.assertEquals(User.Name, sobjField, 'Method must return SObjectField');
+    }
+
+    @IsTest 
+    static void testChainedInvalidField() {
+        Test.startTest();
+        try {
+            FieldReference field = FieldReference.build(Schema.Account.getSObjectType(), 
+                                                    'Account.Name');
+            System.assert(false, 'IllegalArgumentException is expected here');
+        } catch (IllegalArgumentException  e) {
+            System.assertEquals('System.IllegalArgumentException', e.getTypeName());
```

Comment:
What do you test here? e is an instance of IllegalArgumentException by definition